### ISSUE TITLE
form access list / functional tour of Arctos

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#5450]
From ~ https://github.com/ArctosDB/internal/issues/211#issuecomment-1350292164

<img width="776" alt="Screenshot 2023-01-06 at 8 50 05 AM" src="https://user-images.githubusercontent.com/5720791/211058743-d636fe7e-fb3b-47fc-b370-c20a6c4b79ee.png">

Whatever eventually happens in the header/navbar, I think having an annotated list of forms would be super cool so I've started a spreadsheet from the form access control list.

I can't/won't just make the spreadsheet globally editable, but I'm happy to share with anyone who wants access.

https://docs.google.com/spreadsheets/d/1kTTX87YXQynDfdx61yr1ziW7W8K799Wa5kStsx3AinQ/edit#gid=1805650772

My main concern at this point is the shape - would it be useful to add anything beyond description and header_title (which also serves as a flag indicating whether the form may be accessed directly or only through other forms)? 

Here's a sample:

<img width="998" alt="Screenshot 2023-01-06 at 8 46 16 AM" src="https://user-images.githubusercontent.com/5720791/211058482-946d7940-7242-4973-ad41-a69d2f5561f1.png">

HELP!

**Comment by Jegelewicz** [comment]
Is this just a reformat of this? https://arctos.database.museum/Admin/user_roles.cfm

**Comment by Jegelewicz** [comment]
Also, if we do this, can whatever is in the Header/Title and Description also be placed at the top of the associated form?

**Comment by dustymc** [comment]
>eformat of this?

No, that's database roles, this is forms. (But form access is inherited from roles, so there's a strong link.)

This is...

<img width="380" alt="Screenshot 2023-01-06 at 9 10 16 AM" src="https://user-images.githubusercontent.com/5720791/211062414-9fe7b7bd-387f-4392-8af4-b87ee5e835a6.png">

... with more stuff.

>can whatever is in the Header/Title and Description also be placed at the top of the associated form?

Short version: yes.

Longer version: I'll add page help too, the current implementation ain't great. Some of that can be recovered from redirects.

Longer-er version: "Ain't great" includes a ridiculous number of network requests, this will sustain that, maybe ratchet it up a bit. This is one of the many little things which chip away at sustainability, addressing that BEFORE we find ourselves in some scary corner (again) should be a primary function of - uhh, someone. AWG? (Probably not quite?)

**Comment by Jegelewicz** [comment]
OK - my main concern is that we creating definitions for things in multiple places and it won't take more than 5 minutes for them to diverge. See also https://github.com/ArctosDB/internal/issues/225

**Comment by dustymc** [comment]
>creating definitions for things in multiple places 

There's a [word](https://en.wikipedia.org/wiki/Denormalization) for that....

Yes you should be concerned, no I don't think this is replicated anywhere else. There's lots of overlap, but that's just how UI works.

**Comment by dustymc** [comment]
This is going to be in next release, but it's not advertised/linked nor fully integrated yet - the data still need some work, which is becoming increasingly difficult from the disjoined spreadsheet. There will be two levels of access:

https://arctos-test.tacc.utexas.edu/directory.cfm is public. Default as me provides links to lots of forms:


<img width="1237" alt="Screenshot 2023-01-10 at 2 46 23 PM" src="https://user-images.githubusercontent.com/5720791/211678501-e8751c60-1c0e-40a0-aefc-feb15e4e18fb.png">

Or I can filter (un-filter, really) to see things which can't be directly accessed, but might be useful to know about.

<img width="1288" alt="Screenshot 2023-01-10 at 2 46 46 PM" src="https://user-images.githubusercontent.com/5720791/211678598-d2f25464-5650-4539-a68e-d4250baa986a.png">

The public can see the same thing, just with a more-restricted default state.

<img width="887" alt="Screenshot 2023-01-10 at 2 48 08 PM" src="https://user-images.githubusercontent.com/5720791/211678707-c9d5fffd-ae7a-4f84-ac64-b80dea097a9b.png">

https://arctos-test.tacc.utexas.edu/Admin/manage_directory.cfm (which requires global_admin access, as anyone can now see from the directory) provides add and edit access.

TODO

* confirm form access is reasonable
* confirm that everything which can be directly accessed, and nothing which cannot, has a title
* delete the old forms which controlled the simpler table, fix links, close this as complete

TODO, but not part of this (or less critical anyway)

* write form documentation
* check and clean up anything with a deprecated internal comment
* organize directories - that's never really mattered so nothing's been consistent, clear filepaths would help get around this form
* maybe define categories and categorize forms, IDK

**Comment by dustymc** [comment]
rename to tools
expose categories
rebuild UI - category-based lists

**Comment by mkoo** [comment]
@dustymc can you send a spreadsheet of all the possible forms that we need to categorize please?
or we can start with your big menu breakdowns if you send that URL (already forgotten...soemthing 5 was it?)

**Comment by dustymc** [comment]
https://docs.google.com/spreadsheets/d/1kTTX87YXQynDfdx61yr1ziW7W8K799Wa5kStsx3AinQ/edit#gid=1805650772

and I'm not going to get a release out tonight after all, so if the categories can be started by tomorrow I'll try to go out with a simple option rather than changing to add that.

**Comment by mkoo** [comment]
First pass done-- definitely needs work and discussion-- the suggested organization is in the first column, with questions noted. I probably missed some pages that needs a category but I left blank the pages that are not really menu links (like the internal /includes/ or pick cfm's). At least we have a points of discussion now!

**Comment by dustymc** [comment]
Default page:

<img width="1616" alt="Screenshot 2023-01-11 at 4 58 46 AM" src="https://user-images.githubusercontent.com/5720791/211812489-789cbf01-f7a8-4529-a4fb-610f985f35b3.png">

**Comment by ebraker** [comment]
This looks great!

**Comment by dustymc** [comment]
I just revoked access to the spreadsheet - edit can now happen in the Manage Arctos Directory app, which you can find in the [tools directory](https://arctos.database.museum/directory.cfm) (and table view will tell you more).

I've got the cache off for now so you can flip through views to see what happens, but I think it's all pretty intuitive.

**Comment by Jegelewicz** [comment]
And how about the roles currently selected? This one seems weird

![image](https://user-images.githubusercontent.com/5725767/212115273-16df4b6e-2286-4240-87ce-98d9a3c03626.png)

what do containers have to do with that?

**Comment by Jegelewicz** [comment]
Then there is stuff like this

![image](https://user-images.githubusercontent.com/5725767/212115599-c51ad795-ce6d-4efb-b702-ab72a089e6cf.png)

which I guess only works when coming FROM the loan form and so probably should not appear on the tools page?

**Comment by dustymc** [comment]
No, TITLE is what controls links, https://arctos.database.museum/directory.cfm will happily group NULL category. I'll update the top.

Along with the nifty directory, this controls fundamental aspects of Arctos. DO NOT guess at what something does; you can expose things that shouldn't be exposed, make Arctos unusable, etc. (That's why I pushed this out before it's really ready - I ran out of what I could do from the spreadsheet, what's left needs investigation - clicking the link works for some of them, others require digging links out of the code.)

And on that note, a review of https://arctos.database.museum/tools/access_report.cfm?excl_locked=true&role_srch=global_admin&usr_srch= wouldn't hurt.

>what do containers have to do with that?

Nada - exposing data to the sun always reveals some warts, or maybe @mkoo or I did that in preparation for this form. Fixed, thanks.

>probably should not appear on the tools page?

Yes, eventually with a description - so you can flip the tools page to table mode and get an understanding of what Arctos is capable of regarding {category} even if you have to go somewhere else first to get there.

**Comment by Jegelewicz** [comment]
> exposing data to the sun always reveals some warts, or maybe @mkoo or I did that in preparation for this form. Fixed, thanks.

There's quite a few that have manage container, but have nothing to do with object tracking, so maybe a review of those would be helpful? I don't want to change things and screw something up.

> Yes, eventually with a description - so you can flip the tools page to table mode and get an understanding of what Arctos is capable of regarding {category} even if you have to go somewhere else first to get there.

OK, but right now anything with a title shows up and I don't want to add a bunch of things that get you "no", a blank page, or some other error. This list needs to be primarily useful things users can DO.

I think I have the Containers & Barcodes section sort of looking the way it should, but before I continue it would probably help to get a little review.

**Comment by dustymc** [comment]
>quite a few that have manage container

Wilco, that does sound like something I'd do whilst trying to work a spreadsheet...

>anything with a title shows up a

Yes. Everything that's directly accessible should have a title. Nothing which isn't directly accessible should have a title.

>This list needs to be primarily useful things users can DO.

I'm not sure we're clear on this: **THERE ARE MANY LISTS**

"The list" https://arctos.database.museum/directory.cfm?dv= should in fact be limited to things the user in their current state can directly access - things they can DO without going anywhere else first.

"The list" https://arctos.database.museum/directory.cfm?access=all&directaccess=all is everything, most of which won't be accessible to any given user, but might still be helpful in them understanding what Arctos can do (and so what they can do by going through other things first, and what they could do if they'd get the training for more rights, and what they could do if their collection would invest in barcodes, and etc.)

**Comment by Jegelewicz** [comment]
This role definitely seems wrong...

![image](https://user-images.githubusercontent.com/5725767/212131297-9016392a-1f23-4cf2-8c2a-e74c4baf6515.png)

**Comment by dustymc** [comment]
>This role definitely seems wrong...

Elaborate?

The form DOES nothing, so "us" can see it. (But "us" is bigger than we used to be, I'd not object to that being elevated, I'd just like to get some sort of future guidance out of the process.)

**Comment by dustymc** [comment]
containers looks right to me, I did remove title/direct access from a couple that shouldn't have had it.

**Comment by Jegelewicz** [comment]
When I go there I can edit stuff - maybe it looks different when not logged in?

**Comment by Jegelewicz** [comment]
yes, when not logged in I get this

![image](https://user-images.githubusercontent.com/5725767/212133240-d91540ba-8ea0-4026-afae-fed46a64992f.png)

So it seems like one needs more than coldfusion user to access it.

**Comment by dustymc** [comment]
> go there I can edit stuff

Ah - there are three views, see my links a couple issues up - anyone can access /directory

> different when not logged in?

Nope, separate forms at distinct URIs. (Not always the case, but maybe it should be.)

**Comment by dustymc** [comment]
https://arctos.database.museum/directory.cfm?access=all&directaccess=all will tell you what's necessary to access anything.


<img width="1707" alt="Screenshot 2023-01-12 at 9 11 54 AM" src="https://user-images.githubusercontent.com/5720791/212134131-494aa782-8128-42c0-93e5-1163855d21ce.png">

**Comment by dustymc** [comment]
@ewommack I think this could possibly use some sort of announcement. https://arctos.database.museum/directory.cfm will be the primary method of accessing Arctos tools once https://github.com/ArctosDB/internal/issues/211 happens, and that (and https://github.com/ArctosDB/arctos/issues/5333) will happen as soon as I can possibly make them happen. It would be good to handle any change requests while the legacy options are still available.

**Comment by ewommack** [comment]
Ok so what we're looking for is an announcement that says there are big changes afoot for the Arctos Tools that we use for collection management! Please try them out now in case you have any suggestions. Reference this issue and https://github.com/ArctosDB/arctos/issues/5333.
Do I have the summary down correct?

Ways the announcement could go out:
1. Email
2. Newsletter article (which will come out on 31st January)
3. Other social media
Is that what you're looking for @dustymc ?

**Comment by Jegelewicz** [comment]
I've been working on this and just came across these - they both end up at https://arctos.database.museum/Bulkloader/BulkloadSpecimens.cfm

Why?

![image](https://user-images.githubusercontent.com/5725767/212217965-2c2ac82f-24ae-44a9-bb8a-87a181dcce7a.png)

![image](https://user-images.githubusercontent.com/5725767/212218014-8adfe9bb-7d88-448c-b163-e237ccf8879c.png)

**Comment by dustymc** [comment]
>what we're looking 

Yep.

>31st January)

I very much hope that this is done and gone by then. 

>Why

Stability - stuff moves, I try to keep bookmarks working for a while. That's been sorta forever, I'll nuke, thx.

**Comment by Jegelewicz** [comment]
I feel like I at least have the titles somewhat consistent and stuff where I think it goes, but others might have different preferences. I am not really crazy about the "Internal Setting" category - maybe that should just go with Global Admin? I'll look at help links next time.

**Comment by ewommack** [comment]
> 31st January)

> I very much hope that this is done and gone by then.

Then we're looking at an email, and probably tweets. I'm pulling in @lin-fred - @dustymc needs an announcement before the end of next week for how he's going to change the menu items for collection access in Arctos, and we need people to try out the new system and find out what works and what doesn't.

**Comment by dustymc** [comment]
> titles somewhat consistent

Thanks, and I cleaned up a bunch of stuff, and more cooking via Issues. I'm pretty thrilled with this from a code maintenance standpoint, not to mention the fabulous UI.

> "Internal Setting" category - maybe that should just go with Global Admin?

I'd say "Administration" (or Administrative or something) - and I can SQL update if that works.

Maybe merge Community and Profile into - uhhh, something? Not "users" but that's the idea.

Or maybe merge those both into  Agents & Users (under some new name)?

ANYWAY, I'm diggin' it, and it's easy to make changes when someone shows up with clever ideas.

**Comment by dustymc** [comment]
done
